lib/ui/painting.dart

```diff
@@ -144,35 +197,74 @@ class Color {
   /// * Bits 16-23 are the red value.
   /// * Bits 8-15 are the green value.
   /// * Bits 0-7 are the blue value.
-  final int value;
+  @Deprecated('Use component accessors like .r or .g.')
+  int get value =>
+      (_floatToInt8(a) << 24) |
+      (_floatToInt8(r) << 16) |
+      (_floatToInt8(g) << 8) |
+      (_floatToInt8(b) << 0);
 
   /// The alpha channel of this color in an 8 bit value.
   ///
   /// A value of 0 means this color is fully transparent. A value of 255 means
   /// this color is fully opaque.
-  int get alpha => (0xff000000 & value) >> 24;
+  @Deprecated('Use .a.')
+  int get alpha => _floatToInt8(a);
 
   /// The alpha channel of this color as a double.
   ///
   /// A value of 0.0 means this color is fully transparent. A value of 1.0 means
   /// this color is fully opaque.
-  double get opacity => alpha / 0xFF;
+  @Deprecated('Use .a.')
+  double get opacity => a;
 
   /// The red channel of this color in an 8 bit value.
-  int get red => (0x00ff0000 & value) >> 16;
+  @Deprecated('Use .r.')
+  int get red => _floatToInt8(r);
 
   /// The green channel of this color in an 8 bit value.
-  int get green => (0x0000ff00 & value) >> 8;
+  @Deprecated('Use .g.')
+  int get green => _floatToInt8(g);
 
   /// The blue channel of this color in an 8 bit value.
-  int get blue => (0x000000ff & value) >> 0;
+  @Deprecated('Use .b.')
+  int get blue => _floatToInt8(b);
+
+  /// Returns a new color that matches this color with the passed in components
+  /// changed.
+  ///
+  /// Changes to color components will be applied before applying changes to the
+  /// color space.
+  Color withValues(
+      {double? alpha,
+      double? red,
+      double? green,
+      double? blue,
+      ColorSpace? colorSpace}) {
+    Color? tweaked;
+    if (alpha != null || red != null || green != null || blue != null) {
+      tweaked = Color.from(
```

Comment:
Maybe call this something else...

Reply:
What you like, "twisted"?  =). Renamed "updatedComponents".